programs/test/benchmark.c

```diff
@@ -752,13 +868,18 @@ int main( int argc, char *argv[] )
              curve_info++ )
         {
             mbedtls_ecdsa_init( &ecdsa );
-
-            if( mbedtls_ecdsa_genkey( &ecdsa, curve_info->grp_id, myrand, NULL ) != 0 )
-                mbedtls_exit( 1 );
-            ecp_clear_precomputed( &ecdsa.grp );
-
             mbedtls_snprintf( title, sizeof( title ), "ECDSA-%s",
                                               curve_info->name );
+            ret =  mbedtls_ecdsa_genkey( &ecdsa, curve_info->grp_id, myrand, NULL );
+            if( ret  != 0 )
+            {
+                mbedtls_printf( HEADER_FORMAT, title );
+                PRINT_ERROR;
+                continue;
```

Comment:
Should we call `mbedtls_exit( 1 )` here as we did before?

The same goes for other similar places.

Reply:
I prefer not to exit in case of error. Since the code is quite generic, and we don't have a common error code for all feature unavailable, we can't distinguish between a feature unavailable error and a real error, so I would prefer just printing it and go to next step

Reply:
OK, that's a change in behavior. We should do that in its own commit so we can review it and potentially merge it independently from your other changes. Then you can build on top of that change in further commits. I think Andres unified return codes in another PR if not for tests, at least Mbed OS examples; we could do something similar here.

Reply:
ok, I'll return the exit, but this means that when a FEATURE_UNAVAILABLE error returns, we also exit

Reply:
Although it's a change in behaviour, the whole idea of this PR is to support the FEATURE_UNAVAILABLE error, in the applications themselves. If we exit when we get this valid error, it loses the idea of this PR